Preliminary DRAFT for token distribution self custody (#368)
dfinity/docs: modules/token-holders/pages/self-custody.adoc

= Self-custody token distribution
:experimental:
// Define unicode for Apple Command key.
:commandkey: &#8984;
:IC: Internet Computer
:company-id: DFINITY
:sdk-short-name: DFINITY Canister SDK
:sdk-long-name: DFINITY Canister Software Development Kit (SDK)
ifdef::env-github,env-browser[:outfilesuffix:.adoc]

If you recently received notification of your token distribution options and have decided to take ownership and manage the tokens available to you on your own, you can follow the instructions in this guide to authenticate your distribution and take custody of your tokens.

If you aren't sure if self-custody is right for you, be sure to review the information in the link:xxx[Frequently-asked questions] before you continue down this path.

[[before-you-begin]]
== Before you begin

This _Self-Custody Quick Start_ scenario assumes that you are a new token holder and are installing the {sdk-short-name} for the first time on a local computer.

To get start, verify the following:

* You have an internet connection and access to a shell terminal on your local **macOS** or **Linux** computer.

> * You have all of the personal documents—such your passport and driver's license available.

Reviewer: ```suggestion
* You have all of the personal documents — such as your passport and driver's license — available.
```

These documents are required to validate your identity as part of the Know Your Customer (KYC) process.

> * You know how to open a new terminal shell on your local computer and how to run basic command-line programsin the terminal.

Reviewer: ```suggestion
* You know how to open a new terminal shell on your local computer and how to run basic command-line programs in the terminal.
```

If you have a hardware security module (HSM) appliance or a hardware wallet, you can use those devices to generate and store a public/private key pair for your identity in a secure way.
After you generate the public/private key pair you can import your certificate to create a secure identity on a connected device.

If you don't have a hardware security module or a hardware wallet, you can use `+dfx+` to automatically generate and store a public/private key pair for your default identity on the local computer.

== Get the principal for your identity

After you have generated the public/private key pair, you can use dfx to retrieve a principal identifier. A principal identifier is a textual representation of your identity
that is similar to an address in Bitcoin or Ethereum. The principal identifier enables the secret key associated with a user’s public key to be used to sign messages and therefore authenticate actions on your behalf. 

To get the principal:

. Open a terminal shell on your local computer, if you don’t already have one open.
. Check the active identity context by running the following command:
+
[source,bash]
----
dfx identity whoami
----

> . Get the principal for you currently-active identity by running the following command:

Reviewer: ```suggestion
. Get the principal for your currently-active identity by running the following command:
```

+
[source,bash]
----
dfx identity get-principal
----
+

> This command displays the textual representation of your principal as plain text that you can copy and paste.

Reviewer: ```suggestion
This command displays the textual representation of your principal as plain text that you can copy-and-paste.
```
But if you follow https://ell.stackexchange.com/questions/3748/can-i-say-copied-and-pasted, it is just `copy-paste`. Sounds strange to me.

For example:
+
....
g3jww-sbmtm-gxsag-4mecu-72yc4-kef5v-euixq-og2kd-sav2v-p2sb3-pae
....
. Copy the principal identifier for use in the next step.
. Open a web browser and navigate to the link:xxx[Foundation] website.
. Paste the principal you copied in the XX field.
+
If the principal is valid, the Foundation website generates a token-holder identifier (THID) for you to use in the KYC process and you can continue to the next step.
+
If your principal is not validated, submit a link:xx[support request].

== Submit documents for the KYC process

TBD

> After passing the KYC process, yor will receive email notification that your tokens have been released and additional instructions on the next steps, including how to view your distribution and vesting schedule.

Reviewer: ```suggestion
After passing the KYC process, you will receive email notification that your tokens have been released and additional instructions on the next steps, including how to view your distribution and vesting schedule.
```
